FileSystem/FileIO.js

```diff
@@ -0,0 +1,263 @@
+/** 
+  * File open mode enumeration
+  */
+var OpenMode = {
+  APPEND: 'append',
+  BINARY: 'binary',
+  INPUT: 'input',
+  OUTPUT: 'output',
+  RANDOM: 'random'
+}
+
+/** 
+  * File lock mode enumeration
+  */
+var LockMode = {
+  SHARED: 'shared',
+  READ: 'read',
+  WRITE: 'write',
+  READ_WRITE: 'read_write',
+  None: 'none'
+}
+
+/** 
+  * File access mode enumeration
+  */
+var AccessMode = {
+  READ: 'read',
+  WRITE: 'write',
+  READ_WRITE: 'read_write'
+}
+
+/**
+  * File Input Output library
+  */
+var FileIO = {
+  api_open: api_open_,
```

Comment:
Can we name this like openFile. instead?

Google styleguide recommends camel casing for method names -> https://google.github.io/styleguide/jsguide.html#naming

Reply:
Fixed